app/src/main/java/com/bignerdranch/android/criminalintent/CrimeDetailViewModel.kt

```diff
@@ -0,0 +1,26 @@
+package com.bignerdranch.android.criminalintent
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Transformations
+import androidx.lifecycle.ViewModel
+import java.util.UUID
+
+class CrimeDetailViewModel(): ViewModel() {
+
+    private val crimeRepository = CrimeRepository.get()
+    private val crimeIdLiveData = MutableLiveData<UUID>()
+
+    var crimeLiveData: LiveData<Crime?> =
+        Transformations.switchMap(crimeIdLiveData) { crimeId ->
+            crimeRepository.getCrime(crimeId)
+        }
+
+    fun loadCrime(crimeId: UUID) {
```

Comment:
There are two things to consider here:
- like previously, if you are passing an `id` as an argument, it would make sense to call this function `loadCrimeById`
- `loadCrime` implicitly hints that this function has a return type. Think about it, if you were to read that function at the call site, implemented by somebody else, what would you have assumed about it? Considering the fact that the implementation of this method just _sets_ some value to a `LiveData` object, would it make sense to rename this function to `setCrimeById` ?